File: php/sysstatus.php
<?php
// exit if running under command line
if (php_sapi_name() === 'cli') {
    echo("ERROR: Cannot run under command line.".PHP_EOL);
    exit(1);
}

// load libs
require 'libs/system.php';
require 'libs/filesystem.php';
require 'libs/network.php';

$sysstatus = array();

if ($_SERVER['REQUEST_METHOD'] === 'GET') {
    // tags
    $sysstatus['hostname'] = gethostname();
    $sysstatus['time'] = time();
 
    // system metrics
    $sysstatus['system']['loadavg'] = sys_loadavg();
    $sysstatus['system']['file-nr'] = sys_filenr();
    $sysstatus['system']['procs'] = sys_proc();
    $sysstatus['system']['memory'] = sys_mem();
    $sysstatus['system']['cpu'] = sys_cpu();
    $sysstatus['system']['diskperf'] = sys_diskperf();

    // filesystem metrics
    $sysstatus['filesystem']['usage'] = df_usage();
    
    // network metrics
    $sysstatus['network']['netstat'] = net_netstat();
    $sysstatus['network']['netsnmp'] = net_netsnmp();
    $sysstatus['network']['netdev'] = net_netdev();
    
    // encode json
    $sysstatus_json = json_encode($sysstatus);
    
    echo $sysstatus_json;
} else {
    echo "ERROR: HTTP method ".$_SERVER['REQUEST_METHOD']." is not supported.".PHP_EOL;
    http_response_code(400);
}

// debug use
// var_dump($sysstatus);
// var_dump($sysstatus_json);
?>

File: lua/sysstatus.lua
#!/usr/bin/env lua

-- load modules
local json = require "json"
local system = require "system"
local filesystem = require "filesystem"
local network = require "network"
local apache2 = require "apache2"

-- initialize the table
local sysstatus = {}

-- tags
sysstatus["time"] = os.time()

-- system metrics
sysstatus["system"] = {}
sysstatus["system"]["loadavg"] = system.sys_loadavg()
sysstatus["system"]["uptime"] = system.sys_uptime()
sysstatus["system"]["file-nr"] = system.sys_filenr()
sysstatus["system"]["procs"] = system.sys_proc()
sysstatus["system"]["memory"] = system.sys_mem()
sysstatus["system"]["cpu"] = system.sys_cpu()
sysstatus["system"]["diskperf"] = system.sys_diskperf()

-- filesystem metrics
sysstatus["filesystem"] = {}
sysstatus["filesystem"]["usage"] = filesystem.df_usage()

-- network metrics
sysstatus["network"] = {}
sysstatus["network"]["netstat"] = network.net_netstat()
sysstatus["network"]["netsnmp"] = network.net_netsnmp()
sysstatus["network"]["netdev"] = network.net_netdev()

function handle(r)
    r.content_type = "application/json"

    sysstatus["hostname"] = r.server_name
    local sysstatus_json = json.encode(sysstatus)

    if r.method == "GET" then
        r:puts(sysstatus_json)
        r.status = 200
    else
        r:puts("ERROR: HTTP method " .. r.method .. " is not supported.\n")
        r.status = 405
    end

    return apache2.OK
end

-- print(json.encode(sysstatus))
